packages/flutter/lib/src/material_animated_icons/animated_icons.dart

```diff
@@ -79,9 +80,166 @@ class AnimatedIcon extends StatelessWidget {
   /// specified, either directly using this property or using [Directionality].
   final TextDirection textDirection;
 
+  static final _UiPathFactory _pathFactory = () => new ui.Path();
+
   @override
   Widget build(BuildContext context) {
-    // TODO(amirh): implement this.
-    return new Container();
+    // TODO(amirh): implement semantics, text direction, scaling.
+    final _AnimatedIconData iconData = icon;
+    return new CustomPaint(
+      size: iconData.size,
+      painter: new _AnimatedIconPainter(
+        iconData.paths,
+        progress,
+        color,
+        _pathFactory,
+      ),
+    );
+  }
+}
+
+typedef ui.Path _UiPathFactory();
+
+class _AnimatedIconPainter extends CustomPainter {
+  _AnimatedIconPainter(
+    this.paths,
+    this.progress,
+    this.color,
+    this.uiPathFactory,
+  ) : super(repaint: progress);
+
+  // This list is assumed to be immutable, changes to the contents of the list
+  // will not trigger a redraw as shouldRepaint will keep returning false.
+  final List<_PathFrames> paths;
+  final Animation<double> progress;
```

Comment:
you could save some memory by having progress here just redirect to the listenable in the parent

Reply:
Not sure I follow, do you mean something like:
Animation<double> get progress => super._repaint;

Can I do that? isn't _repaint is private to the rendering library?
Also, is it wrong to think that this kind of optimization (saving a single reference) is negligible in this context?
Is there any cost to adding a getter?
Is there any chance that the compiler is able to do this optimization on it's own (given that the member is final) ?

Reply:
Oh, hey, we didn't make that public. Go figure. Disregard my suggestion!

> is it wrong to think that this kind of optimization (saving a single reference) is negligible in this context?

It's not wrong. However, trivial optimisations add up. If there's no cost to doing them, then on the long run you're better off if you avoid the cost of not doing them.

> Is there any cost to adding a getter?

There's a memory cost at the class level (vs the instance level), and a minor cost per access, though that cost should be optimised away.

> Is there any chance that the compiler is able to do this optimization on it's own

I wouldn't count on it.